Revamp Permalink section

rlue opened the pull request:
The previous version contained errors regarding the definition and usage of the `:path` permalink template variable (see Issue #5901). This version corrects those errors, removes redundant language where possible, and restructures the section to place examples before template variable definitions.

The reasoning behind the third change is that it's generally easier for learners to understand a generalized or abstracted definition after having seen concrete examples in action first.

parkr commented:
/cc @jekyll/documentation

tomjohnson1492 commented on docs/_docs/collections.md:
````diff
@@ -89,18 +89,59 @@ choice and written out to `<dest>/my_collection/some_subdir/some_doc.html`.
+```
+├── _apidocs
+│   └── archive
+│       └── deprecated.md
````
Why do you have "deprecated.md" as an example of the file here? What's this about?

rlue replied:
Ha, I was worried that might raise an eyebrow. 

I figured that a generic-but-realistic use case would make for a more descriptive example than `apidocs/mydocs/doc1.md`; _i.e.,_ that it would be easier to imagine why you might choose to set your permalinks up as, say, `doc/deprecated.html` than `awesome/doc1.html`. But I also wrote it in a hurry, and I recognize that maybe `apidocs/archive/deprecated.md` is not as realistic as it could be.

If it creates more confusion than it resolves, I'm happy to change it back to `apidocs/mydocs/doc1.md` and resubmit the PR.

tomjohnson1492 replied:
I suggest using a generic file name that doesn't call attention to itself like "deprecated" does. Maybe myfile.my or something. I have some more comments that I'll insert inline.

DirtyF replied:
We *could* use a more generic file name, but I don't mind the current example here, seems pretty explanatory to me.

tomjohnson1492 commented on docs/_docs/collections.md:
```diff
@@ -89,18 +89,59 @@ choice and written out to `<dest>/my_collection/some_subdir/some_doc.html`.
 ## Configuring permalinks for collections {#permalinks}
 
-You can customize the [Permalinks](../permalinks/) for your collection's documents by setting `permalink` property in the collection's configuration as follows:
+You can specify a pattern for the URLs where your Collection documents will reside with the [`permalink` property](../permalinks/):
```
There's a modifier problem with the last "with the ..." clause. Maybe recast as "Through the [`permalink` property](../permalinks/), you can specify a pattern for the URLs for your Collection documents. The permalink also defines the structure for the files in the output."

tomjohnson1492 commented:
In the Examples section, the tree formatting is messed up. This may be intentional, but it looks awkward like this to me. Can you preserve the tree formatting in the examples in a way that leaves the tree diagrams complete?

tomjohnson1492 commented:
If you don't want to use the complete tree diagrams there, you could use forward slashes instead.

parkr commented:
@DirtyF Where are you with this? Need more info?

rlue commented:
Sorry to ghost you guys, I'm happy to make the changes @tomjohnson1492 requested, and will get to work on it this weekend.

rlue commented:
Okay folks, I've made the changes requested.

@tomjohnson1492:

* In the interest of consistency, I abandoned the `apidocs` example Collection altogether in favor of `_my_collection/some_subdir/some_doc.md` (which appears in [the section immediately preceding Permalinks](https://jekyllrb.com/docs/collections/#step3))
* Thanks for pointing out the misplaced modifier. Hopefully the new version is to your liking?
* Not sure exactly what was wrong with the tree diagrams in my last commit; I simply copied the format as it was [before my edits](https://jekyllrb.com/docs/collections/#permalink-examples-for-collections). I suspect you either meant that root directories for the tree diagrams were absent, or that in the first column, vertical lines descended down from the top and didn't connect to anything underneath.

  To address the first point, I added root directories to each tree diagram. To address the second, I added ellipses to the bottom of some tree diagrams to indicate that there were irrelevant portions omitted.

Let me know if there's anything else I should do!

DirtyF commented on docs/_docs/collections.md:
````diff
@@ -89,18 +89,70 @@ choice and written out to `<dest>/my_collection/some_subdir/some_doc.html`.
 ```yaml
 collections:
-  my_collection:
+  categories:
````
Did you mean to replace `my_collection` with `categories`here?

DirtyF commented on docs/_docs/collections.md:
````diff
@@ -89,18 +89,70 @@ choice and written out to `<dest>/my_collection/some_subdir/some_doc.html`.
+
+* **Default**  
+  Same as `permalink: /:collection/:path`.
+  ```
````
Please insert carriage returns to create a paragraph before each code sample

```
*  **Default**

Same as `permalink: /:collection/:path`.

```

rlue commented:
Good looking out!

DirtyF commented:
@rlue Thanks for making Jekyll's documentation better 📗 

@jekyllbot: merge +docs
